### PR TITLE
[Merged by Bors] - chore(AlgebraicTopology/SingularSet): factor topology out of `SimplicialSet` into new file

### PR DESCRIPTION
**Pull request by jcommelin**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/AlgebraicTopology/SingularSet.lean [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/-
+Copyright (c) 2024 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin, Scott Morrison, Adam Topaz
```
Comment (YaelDillies) [inline Mathlib/AlgebraicTopology/SingularSet.lean:4]: Can you make sure the year is correct?

**Review by YaelDillies — COMMENTED** [review]
I'm happy with this but an algebraic topology maintainer (@adamtopaz ?) should have a look.

File: Mathlib/AlgebraicTopology/SingularSet.lean (hunk 1 above)
Comment (jcommelin) [reply Mathlib/AlgebraicTopology/SingularSet.lean:4]: Shouldn't it be the year that this file was created?

Comment (YaelDillies) [reply Mathlib/AlgebraicTopology/SingularSet.lean:4]: Ah, I guess [we've argued this before](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Copyright.20header/near/280420005). I always take it to be the year the oldest definition/lemma of the file was created.

File: Mathlib/AlgebraicTopology/SingularSet.lean [hunk 2]
```diff
@@ -0,0 +1,71 @@
+/-
+Copyright (c) 2024 Johan Commelin. All rights reserved.
```
Comment (jcommelin) [inline Mathlib/AlgebraicTopology/SingularSet.lean:2]:
```suggestion
Copyright (c) 2023 Johan Commelin. All rights reserved.
```

**Comment by YaelDillies** [comment]
Please an algebraic topology maintainer sign off!

maintainer merge

**Comment by joelriou** [comment]
This looks good to me. There is one issue though (which can fixed in a later PR): it seems that the functors like `TopCat.toSSet : TopCat ⥤ SSet` are currently defined only for topological spaces in `Type 0`.

bors d+

**Pull request closed without merging**